Source/WebKit/UIProcess/WebProcessPool.cpp

```diff
@@ -1947,7 +1947,7 @@ std::tuple<Ref<WebProcessProxy>, SuspendedPageProxy*, ASCIILiteral> WebProcessPo
     if (m_automationSession)
         return { WTFMove(sourceProcess), nullptr, "An automation session is active"_s };
 
-    if (!sourceProcess->hasCommittedAnyProvisionalLoads()) {
+    if (!sourceProcess->hasCommittedAnyProvisionalLoads() && !page.preferences().siteIsolationEnabled()) {
```

Comment:
Why are we disabling this optimization? This seems bad. I don't think this optimization is incompatible with site isolation.